unstract/connectors/src/unstract/connectors/databases/exceptions.py

```diff
@@ -0,0 +1,84 @@
+from typing import Any
+
+from rest_framework.exceptions import APIException
+
+
+class UnstractDBConnectorException(APIException):
+    status_code = 400
+    default_detail = "Error creating/inserting to database. "
+
+    def __init__(self, code: Any = status_code, detail: str = default_detail) -> None:
+        super().__init__(code=code, detail=detail)
```

Comment:
@kirtimanmishrazipstack `unstract-connectors` is a standalone python library that could be used with django / flask or any other means as well. We shouldn't be opinionated here and inherit from `APIException`. Please move this piece to the backend itself where API related exceptions are raised.

However you can have a base connector error class - possibly specialized for DBs which can further be inherited. Refer [how this ConnectorError is created / used](https://github.com/Zipstack/unstract/blob/26ebb17d47260f73e4e1f931ccf141f082f4f380/unstract/connectors/src/unstract/connectors/exceptions.py#L18-L33)

As a result, I don't think the status code itself should be passed from this library since it can be used independent of an API as well - in such cases, these exceptions won't make much sense.
One option is to have a dedicated `ConnectorAPIException` class which can be inherited - but I'm not a huge fan of this approach and its debatable

Reply:
Implemented